src/battlegoals/partyBattleGoals.tsx

```diff
@@ -2,78 +2,177 @@ import * as React from "react";
 import { NoProps } from "lang/react";
 import { partition, shuffle } from "lang/arrays";
 import { range } from "lang/ranges";
-import PlayerBattleGoals from "battlegoals/playerBattleGoals";
-import CommunityBattleGoalCard from "battlegoals/communityBattleGoalCard";
-import BattleGoalCard from "./battleGoalCard";
+import { BattleGoal, battleGoalByGlobalId, communityBattleGoals, officialBattleGoals } from "battlegoals/battleGoals";
+import BattleGoalCard from "battlegoals/battleGoalCard";
 
-function battleGoals(): Array<number> {
-  return range(1, 24);
-}
-
-function communityBattleGoals(): Array<number> {
-  return range(1, 100);
-}
-
-function drawDistinctBattleGoals(allBattleGoals: Array<number>, count:number): Array<number> {
+function drawDistinctBattleGoals(allBattleGoals: Array<BattleGoal>, count: number): Array<BattleGoal> {
   return shuffle(allBattleGoals).slice(0, count);
 }
 
-enum BattleGoalFlavour {
-  Vanilla = 'Vanilla',
-  Community = 'Community',
+interface Picks {
+  [picks: number]: number
 }
 
 interface PartyBattleGoalsState {
-  flavour: BattleGoalFlavour;
+  includeVanilla: boolean;
+  includeCommunity: boolean;
+  drawnBattleGoals: BattleGoal[];
+  currentPlayer: void | number;
+  picks: Picks;
 }
 
 export default class PartyBattleGoals extends React.Component<NoProps, PartyBattleGoalsState> {
+  private storage: Storage = window.localStorage;
 
   constructor(props: NoProps) {
     super(props);
-    this.handleFlavourChange = this.handleFlavourChange.bind(this);
+    this.handleDrawBattleGoals = this.handleDrawBattleGoals.bind(this);
+    this.toggleVanilla = this.toggleVanilla.bind(this);
+    this.toggleCommunity = this.toggleCommunity.bind(this);
+    this.logState = this.logState.bind(this);
+    this.handlePlayerToggle = this.handlePlayerToggle.bind(this);
+    this.handlePlayerPick = this.handlePlayerPick.bind(this);
     this.state = {
-      flavour: BattleGoalFlavour.Vanilla
+      includeVanilla: false,
+      includeCommunity: false,
+      drawnBattleGoals: [],
+      currentPlayer: undefined,
+      picks: {}
     }
   }
 
-  private handleFlavourChange(event: React.ChangeEvent<HTMLSelectElement>) {
-    this.setState({ flavour: event.target.value as BattleGoalFlavour }, () => {
-      console.log(this.state)
-    });
+  componentDidMount() {
+    const dtoAsString = this.storage.getItem('partyBattleGoalState');
```

Comment:
dto = data transfer object?

That's a java-ism but I'll allow it this once ;)

Reply:
we can call it `storageFormat` if you want :)

Reply:
btw. this is not necessary java related https://en.wikipedia.org/wiki/Data_transfer_object

Reply:
Wikipedia cites MSDN and Fowler, so I guess it's a C# / Java-ism :-P

Reply:
Feel free to keep the name.

Reply:
:)